lib/logstash/inputs/couchdb_changes.rb

```diff
@@ -0,0 +1,217 @@
+# encoding: utf-8
+
+require "logstash/inputs/base"
+require "logstash/namespace"
+require "net/http"
+require "uri"
+
+# Stream events from the CouchDB _changes URI.
+# Use event metadata to allow for upsert and
+# document deletion.
+class LogStash::Inputs::CouchDBChanges < LogStash::Inputs::Base
+  config_name "couchdb_changes"
+  milestone 1
+
+  # IP or hostname of your CouchDB instance
+  # Default: localhost
+  config :host, :validate => :string, :default => "localhost"
+
+  # Port of your CouchDB instance
+  # Default: 5984
+  config :port, :validate => :number, :default => 5984
+
+  # The CouchDB db to connect to.
+  # Required parameter.
+  config :db, :validate => :string, :required => true
+
+  # Connect to CouchDB's _changes feed securely (via https)
+  # Default: false (via http)
+  config :secure, :validate => :boolean, :default => false
+  
+  # Path to a CA certificate file, used to validate certificates
+  config :ca_file, :validate => :path
+
+  # Username, if authentication is needed to connect to 
+  # CouchDB
+  config :username, :validate => :string, :default => nil
+
+  # Password, if authentication is needed to connect to 
+  # CouchDB
+  config :password, :validate => :string, :default => nil
+  
+  # Logstash connects to CouchDB's _changes with feed=continuous
+  # The heartbeat is how often (in milliseconds) Logstash will ping
+  # CouchDB to ensure the connection is maintained.  Changing this 
+  # setting is not recommended unless you know what you are doing.
+  config :heartbeat, :validate => :number, :default => 1000
```

Comment:
If we don't want to support "duration units" (like 3ms), then for backwards compatibility (in the future) we'll want 'heartbeat' to be in seconds. THis way, when we add units later, a plain number with no unit can still mean seconds - it would confuse me if 'heartbeat' supported units but a unitless number meant "milliseconds" :P